### DISCUSSION TITLE
Window.close not working

### DISCUSSION THREAD

**Discussion by pwpretz** [PrismarineJS/mineflayer#3150] [Q&A]
So im trying to make a bot sell stuff to the shop and it wont exit the gui

my code is 

bot.chat("/ ds qsell")
  bot.on("windowOpen", (window) => {
    bot.inventory.slots.forEach((item) => {
      if (item) {
          window.deposit(859, null, 64)
      }
    })
  window.close()
    console.log("sold")
  })

after some time off staying in the gui it gives me this error

FATAL ERROR: Ineffective mark-compacts near heap limit Allocation failed - JavaScript heap out of memory

**Comment by extremeheat** [comment]
> JavaScript heap out of memory

Means you have a memory leak somewhere. You need to rewrite your program to fix this issue, or if you don't know what you're doing, start clean with an example then see what code that you add triggers the issue.
